scuttlebutt/src/lib.rs

```diff
@@ -39,15 +39,53 @@ use tracing::{debug, error};
 use crate::digest::Digest;
 use crate::message::ScuttleButtMessage;
 use crate::serialize::Serializable;
-pub use crate::state::ClusterState;
+pub use crate::state::{ClusterState, SerializableClusterState};
 use crate::state::NodeState;
 
 /// Map key for the heartbeat node value.
 pub(crate) const HEARTBEAT_KEY: &str = "heartbeat";
 
 pub type Version = u64;
 
-pub type NodeId = String;
+// pub type NodeId = String;
+
+#[derive(Default, Clone, Hash, Eq, PartialEq, PartialOrd, Ord, Debug, Serialize, Deserialize)]
+pub struct NodeId {
+    id: String,
+    gossip_public_address: String,
```

Comment:
why not SocketAddr ?

Reply:
hmm, I needed to implement Default, not sure of a good default SocketAddr and this is not an option really.  Also, this made testing easier because I could `NodeId::from("node-one")` without worrying of passing a valid SocketAdrr.